src/Language/Haskell/TH/Instances.hs

```diff
@@ -403,18 +405,26 @@ deriving instance Bounded Extension
 $(deriveQuasiTrans
     [t| forall r m. Quasi m => Proxy2 (ReaderT r m) |]
     [e| \m1 m2 -> ReaderT $ \ r -> runReaderT m1 r `qRecover` runReaderT m2 r |])
+$(deriveQuoteTrans
```

Comment:
I think Quote isntance should come *before* Quasi instance. Does this work with GHC-9.0, where TH splices and instances interaction is stricter.

Reply:
I'm fine with this suggestion, and I've pushed a change with applies it. It is worth noting, however, that `Quote` is _not_ a superclass of `Quasi`, so GHC 9.0 is fine with either order.